src/config/mod.rs

```diff
@@ -0,0 +1,1221 @@
+                    if let Some(p) = OptionalPolicy::try_from(pe)? {
+                        config.policies.push(p);
+                    }
+                }
+                Element::Servicedir(p) => {
+                    config.servicedirs.push(p);
+                }
+                Element::Servicehelper(p) => {
+                    // NOTE: we're assuming this has the same "last one wins" behaviour as `<type>`
+
+                    // TODO: warn and then ignore if we aren't reading:
+                    // /usr/share/dbus-1/system.conf
+                    config.servicehelper = Some(p);
+                }
+                Element::StandardSessionServicedirs => {
+                    // TODO: warn and then ignore if we aren't reading: /etc/dbus-1/session.conf
+                    if let Ok(runtime_dir) = var("XDG_RUNTIME_DIR") {
+                        config
+                            .servicedirs
+                            .push(PathBuf::from(runtime_dir).join("dbus-1/services"));
+                    }
+                    if let Ok(data_dir) = var("XDG_DATA_HOME") {
+                        config
+                            .servicedirs
+                            .push(PathBuf::from(data_dir).join("dbus-1/services"));
+                    }
+                    let mut servicedirs_in_data_dirs = xdg_data_dirs()
+                        .iter()
+                        .map(|p| p.join("dbus-1/services"))
+                        .map(PathBuf::from)
+                        .collect();
+                    config.servicedirs.append(&mut servicedirs_in_data_dirs);
+                    config
+                        .servicedirs
+                        .push(PathBuf::from("/usr/share/dbus-1/services"));
+                    // TODO: add Windows-specific session directories
+                }
+                Element::StandardSystemServicedirs => {
+                    // TODO: warn and then ignore if we aren't reading:
+                    // /usr/share/dbus-1/system.conf
+                    config
+                        .servicedirs
+                        .extend(STANDARD_SYSTEM_SERVICEDIRS.iter().map(PathBuf::from));
+                }
+                Element::Syslog => config.syslog = true,
+                Element::Type(TypeElement { r#type: value }) => config.r#type = Some(value),
+                Element::User(s) => config.user = Some(s),
+            }
+        }
+
+        Ok(config)
+    }
+}
+
+impl Config {
+    pub fn parse(s: &str) -> Result<Self> {
+        // TODO: validate that our DOCTYPE and root element are correct
+        quick_xml::de::from_str::<Document>(s)?.try_into()
+    }
+
+    pub fn read_file(file_path: impl AsRef<Path>) -> Result<Self> {
+        // TODO: error message should contain file path to missing `<include>`
+        Document::read_file(&file_path)?.try_into()
+    }
+}
+
+#[derive(Clone, Debug, Deserialize, PartialEq)]
+pub struct ConnectOperation {
```

Comment:
See a lot of types here. Why not split them into their own file now that `config` is a directory?